frontend/lib/src/components/elements/Expander/Expander.tsx

```diff
@@ -73,29 +76,19 @@ export const ExpanderIcon = (props: ExpanderIconProps): ReactElement => {
         {...iconProps}
       />
     )
-  } else if (icon === "check") {
-    return (
-      <StyledIcon
-        as={Check}
-        color={"inherit"}
-        aria-hidden="true"
-        data-testid="stExpanderIconCheck"
-        {...iconProps}
-      />
-    )
-  } else if (icon === "error") {
-    return (
-      <StyledIcon
-        as={ErrorOutline}
-        color={"inherit"}
-        aria-hidden="true"
-        data-testid="stExpanderIconError"
-        {...iconProps}
-      />
-    )
   }
 
-  return <></>
+  return icon ? (
+    <DynamicIcon
+      color="inherit"
+      iconValue={icon}
+      aria-hidden={statusIconTestIds[icon] ? "true" : undefined}
+      data-testid={statusIconTestIds[icon] || "stExpanderIcon"}
```

Comment:
It think we better should use the `testid` prop defined in DynamicIcon here instead. Using other props here is working since we are routing it through to the styled component, but that's a very unusual pattern (maybe we should even remove this part). It is better torelyy on the props that are explicitly defined in DynamicIcon.

Reply:
Good call! I saw the prop forwarding being done on `StyledIcon` and didn't question if it was unusual. But I've removed that in `StyledDynamicIcon` and made it rely on props that are explicitly defined in DynamicIcon 👍 Also, made the switch from `data-testid` to the `testid` prop defined in DynamicIcon